lib/circuit/CircuitStateStorage/ArgumentsCircuitStateStorage.ts

```diff
@@ -0,0 +1,22 @@
+import * as hash from 'object-hash';
+
+import { CircuitState } from '../CircuitState/CircuitState';
+import { CircuitStateFactory } from '../CircuitState/factory';
+import { CircuitStateStorage } from './CircuitStateStorage';
+
+export class ArgumentsCircuitStateStorage implements CircuitStateStorage {
+
+  private readonly argumentsStorage = new Map<string, CircuitState>();
+
+  constructor(private readonly circuitStateFactory: CircuitStateFactory) { }
+
+  public get(args: any[]): CircuitState {
+    const key = hash(args);
```

Comment:
Later, when merged with `@cache` this has to be refactored to use the `HashService` from `/utils`